packages/bundlers/default/src/DefaultBundler.js

```diff
@@ -16,6 +16,33 @@ import nullthrows from 'nullthrows';
 import path from 'path';
 import {encodeJSONKeyComponent} from '@parcel/diagnostic';
 
+/**
+ * A chained set propagates `add` to the parent but not the other way around
+ */
+class ChainedSet<T> {
```

Comment:
Worker problem solved by using `ChainedSet` (so that adding a sibling bundle for a parent doesn't propagate down, only adding a sibling to a child node should add it to the ancestry as well)

Reply:
The more we add to this, the more I think maybe it should be part of the graph?